torchtitan/checkpoint.py

```diff
@@ -123,6 +123,30 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         list(map(func, self.model, self.optim))
 
 
+class OptimizerInBackwardWrapper(OptimizerWrapper):
```

Comment:
I'm not sure if we want another wrapper. Can't we just flatten the `OptimizerContainer` in `OptimizerWrapper.__init__`?

Reply:
Flaten optims here so could follow the same get/set state_dict logic